pylabrobot/centrifuge/vspin.py

```diff
@@ -39,110 +39,111 @@ async def setup(self):
     self.dev.open()
     logger.debug("open")
     # TODO: add functionality where if robot has been intialized before nothing needs to happen
-    for _ in range(3):
-      await self.configure_and_initialize()
+    try:
+      for _ in range(3):
+        await self.configure_and_initialize()
+        await self.send(b"\xaa\x00\x21\x01\xff\x21")
       await self.send(b"\xaa\x00\x21\x01\xff\x21")
-    await self.send(b"\xaa\x00\x21\x01\xff\x21")
-    await self.send(b"\xaa\x01\x13\x20\x34")
-    await self.send(b"\xaa\x00\x21\x02\xff\x22")
-    await self.send(b"\xaa\x02\x13\x20\x35")
-    await self.send(b"\xaa\x00\x21\x03\xff\x23")
-    await self.send(b"\xaa\xff\x1a\x14\x2d")
-
-    self.dev.baudrate = 57600
-    self.dev.ftdi_fn.ftdi_setrts(1)
-    self.dev.ftdi_fn.ftdi_setdtr(1)
-
-    await self.send(b"\xaa\x01\x0e\x0f")
-    await self.send(b"\xaa\x01\x12\x1f\x32")
-    for _ in range(8):
-      await self.send(b"\xaa\x02\x20\xff\x0f\x30")
-    await self.send(b"\xaa\x02\x20\xdf\x0f\x10")
-    await self.send(b"\xaa\x02\x20\xdf\x0e\x0f")
-    await self.send(b"\xaa\x02\x20\xdf\x0c\x0d")
-    await self.send(b"\xaa\x02\x20\xdf\x08\x09")
-    for _ in range(4):
-      await self.send(b"\xaa\x02\x26\x00\x00\x28")
-    await self.send(b"\xaa\x02\x12\x03\x17")
-    for _ in range(5):
-      await self.send(b"\xaa\x02\x26\x20\x00\x48")
+      await self.send(b"\xaa\x01\x13\x20\x34")
+      await self.send(b"\xaa\x00\x21\x02\xff\x22")
+      await self.send(b"\xaa\x02\x13\x20\x35")
+      await self.send(b"\xaa\x00\x21\x03\xff\x23")
+      await self.send(b"\xaa\xff\x1a\x14\x2d")
+      self.dev.baudrate = 57600
+      self.dev.ftdi_fn.ftdi_setrts(1)
+      self.dev.ftdi_fn.ftdi_setdtr(1)
+      self.get_status()
+    except:
```

Comment:
what kind of error is raised? / why is `except` needed?

Reply:
In status_check, an error is raised when nothing is returned from the status check. So if nothing is returned the centrifuge isn't init and the error will trigger the except, which is the actual init